backend/api/serializers/ClassSerializer.py

```diff
@@ -1,13 +1,28 @@
+import random
+import string
 from rest_framework import serializers
 
 from api.models import Class
 
 class ClassesSerializer(serializers.ModelSerializer):
     class Meta:
         model = Class
-        fields = ['id', 'name', 'sections', 'schedule']
+        fields = ['id', 'teacherId', 'name', 'sections', 'schedule', 'class_code']
         labels = {
             'name': 'Class Name',
+            'teacherId': 'Teacher',
             'sections': 'Number of Sections',
             'schedule': 'Schedule',
-        }
\ No newline at end of file
+            'class_code': 'Class Code'
+        }
+
+    def create(self, validated_data):
```

Comment:
should checks if class_code doesn't exist in the database

Reply:
All requested changes are made. You may review them now.